### ISSUE TITLE
`polars` operations with columns in lazyframes

### ISSUE THREAD

**Issue by maxim-uvarov** [nushell/nushell#12809]
### Describe the bug

After using `polars rename` error happens when trying to substract one column from another.

### How to reproduce

```
❯ let new = ([9 8 4] | polars into-df)

> let new_2 = $new * 3 + 10 | polars rename "0" memorable

> $new_2
╭────────────────┬────────────────────────────────────────────────────╮
│ plan           │ RENAME                                             │
│                │   DF ["0"]; PROJECT */1 COLUMNS; SELECTION: "None" │
│ optimized_plan │ RENAME                                             │
│                │   DF ["0"]; PROJECT */1 COLUMNS; SELECTION: "None" │
╰────────────────┴────────────────────────────────────────────────────╯

> $new - $new_2
Error: nu::shell::cant_convert

  × Can't convert to NuDataFrame.
   ╭─[entry #18:1:8]
 1 │ $new - $new_2
   ·        ───┬──
   ·           ╰── can't convert NuLazyFrameCustomValue to NuDataFrame
   ╰──── 
```

### Expected behavior

I expect `polars` to work

### Configuration

| key                | value                                                      |
| ------------------ | ---------------------------------------------------------- |
| version            | 0.93.1                                                     |
| major              | 0                                                          |
| minor              | 93                                                         |
| patch              | 1                                                          |
| branch             | main                                                       |
| commit_hash        | c54d223ea0b14f180b1793fa539419854c2e014f                   |
| build_os           | macos-aarch64                                              |
| build_target       | aarch64-apple-darwin                                       |
| rust_version       | rustc 1.77.2 (25ef9e3d8 2024-04-09)                        |
| rust_channel       | 1.77.2-aarch64-apple-darwin                                |
| cargo_version      | cargo 1.77.2 (e52e36006 2024-03-26)                        |
| build_time         | 2024-05-07 05:50:32 +00:00                                 |
| build_rust_channel | release                                                    |
| allocator          | mimalloc                                                   |
| features           | dataframe, default, sqlite, system-clipboard, trash, which |
| installed_plugins  | explore, image, inc, polars                                |

**Comment by ayax79** [comment]
I did an initial investigation. It looks like I need to do some work around operations for dataframes. This should be considered part of the lazy frame refactor work.

**Comment by ayax79** [comment]
I create a pull request for the fix: #12819

**Comment by maxim-uvarov** [comment]
I'm not sure if I should create a new issue for this. Please let me know if I should. 

This kind of operation should be supported, as I understand:

```
> let new = [9 8 4] | polars into-df
> let mask = $new == 8
> $mask | polars expr-not
Error: nu::parser::input_type_mismatch

  × Command does not support dataframe input.
   ╭─[entry #45:1:9]
 1 │ $mask | polars expr-not
   ·         ───────┬───────
   ·                ╰── command doesn't support dataframe input
   ╰────
```
